### PR TITLE
Add unified logging framework

### PR DESCRIPTION
**Pull request by panyu4**
Signed-off-by: panyu4 <PY.panyu@huawei.com>

### CODE REVIEW

File: src/comm/util.js [hunk 1]
```diff
@@ -14,10 +14,146 @@
...
+ * @return {object}   the new winston logger
+ */
+function newDefaultLogger() {
+    let transports =[new (winston.transports.Console)({
+        colorize: true})];
+    ['debug', 'info', 'warn', 'error'].forEach((level)=>{
+        try {
+            //remove the last file;
+            deleteLoggerFiles(level);
+            if(level==='debug'){
```
Comment (aklenik) [inline src/comm/util.js]: I think the default should be something like the following:
* info (which is essentially progress report) and above (warning, error) should be printed to the console
* debug should go to a debug file

File: benchmark/simple/config.json [hunk 2]
```diff
@@ -2,11 +2,17 @@
   "blockchain": {
     "type": "fabric",
     "config": "benchmark/simple/fabric.json"
-  },
+  },  
   "command" : {
     "start": "docker-compose -f network/fabric/simplenetwork/docker-compose.yaml up -d",
     "end" : "docker-compose -f network/fabric/simplenetwork/docker-compose.yaml down;docker rm $(docker ps -aq);docker rmi $(docker images dev* -q)"
   },
+  "logging": {
```
[the remark this answers is not in the document]
Comment (panyu4) [reply benchmark/simple/config.json]: I've moved these setting to default.yaml

File: src/comm/bench-flow.js [hunk 3]
```diff
@@ -22,14 +22,15 @@ const Monitor = require('./monitor.js');
 const Report  = require('./report.js');
 const Client  = require('./client/client.js');
 const Util = require('./util.js');
-const log = Util.log;
+const logger = Util.getLogger('testLog');
```
[the remark this answers is not in the document]
Comment (panyu4) [reply src/comm/bench-flow.js]: All of the logger names are modified to the corresponding file name.

File: src/comm/util.js [hunk 4]
```diff
@@ -14,10 +14,146 @@
...
+                        json: false,
+                        maxsize: maxSize,
+                        handleExceptions: true,
+                        formatter: logFormat
+                    })
+                );
+            }
+        }
+        catch (ex) {
+            throw new Error('failed to set transports for files, the logger will use the default setting ');
```
[the remark this answers is not in the document]
Comment (panyu4) [reply src/comm/util.js]: It is fixed.

**Comment by aklenik** [comment]
@panyu4 Some suggestions before merging:

1. There are parts of the code-base that is executed many-many times. For example, callbacks and rate controllers are called for every client instance for every tx. The info logging in these should be changed to debug level, otherwise, the user will see every kind of (not really update) message, and won't really see the Tx progress. Hopefully, there aren't many callbacks and rate controllers, so this is easy to change.

1. The console transport of the default logger is created without a level. It should only write info and above (so not debug). The file logger can log everything.

1. Some files read the logger settings, then pass it to the `getLogger` method. I think the logger settings should be read inside the `getLogger` method only, consumers of the logger shouldn't care about the settings, they just have to use the logger.

**Comment by panyu4** [comment]
@aklenik , Thanks. I've finished the modifications according to your suggestions. Please review again.

**Comment by aklenik** [comment]
@panyu4 When did you make the last modification? Because I just finished reviewing it. Did you make changes in the last 15 minutes? :)

**Pull request merged**